File: frontend/checkoutcalculator/src/model/Order.ts
import { Item } from "@/model/Item";

export class Order {
  uuid: string;
  orderList: Item[];
  client: string;
  date: string;

  constructor(uuid= "",
              client= "Unknown Client",
              list: Item[] = [],
              data= "") {
    this.uuid = uuid;
    this.client = client;
    this.orderList = list;
    this.date = data
  }

  DropQ(item: Item): void{
    item.quantity -= 1;
  }
  AddQ(item: Item): void{
    item.quantity += 1;
  }
  AddItem(item: Item): void{
    this.orderList.push(item);
  }
  DelItem(item: Item): void{
    this.orderList = this.orderList.filter(value => value != item);
  }
}

File: backend/src/main/java/com/csc301/chckout/chechoutAPI/Entity/Order.java
package com.csc301.chckout.chechoutAPI.Entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
public class Order {
    @JsonProperty(value = "uuid")
    private String uuid;
    @JsonProperty(value = "orderList")
    private List<Item> orderList;
    @JsonProperty(value = "client")
    private String client = "Unknown Client";
    @JsonProperty(value = "date")
    private String date = "";
}
